e2e/_suites/metricbeat/metricbeat_test.go

```diff
@@ -406,10 +406,18 @@ func (mts *MetricbeatTestSuite) runMetricbeatService() error {
 			return err
 		}
 
+		mts.Version = mts.Version + "-amd64"
+
+		// wait for tagging to ensure the loaded image is present
+		e2e.Sleep(3 * time.Second)
```

Comment:
Otherwise, it could be the case that the process go faster that the docker engine to list the previously loaded image

Reply:
Instead of adding a sleep could we just poll the docker api to verify the image is present? It may be useful to abstract some api's that could be reused in different areas and hopefully provide a better guarantee of making sure these requirements are met before proceeding. Thoughts?

Reply:
That would be lovely! Because this code (the tag image code) is very isolated and narrowed down to two usages, let's merge it as is and create an issue for that using the Docker client API.